src/main/java/ru/yandex/practicum/filmorate/service/UserService.java

```diff
@@ -0,0 +1,54 @@
+package ru.yandex.practicum.filmorate.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import ru.yandex.practicum.filmorate.exceptions.NonexistentException;
+import ru.yandex.practicum.filmorate.model.User;
+import ru.yandex.practicum.filmorate.storage.UserStorage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserService {
+    private final UserStorage userStorage;
+
+    public void addFriend(User user, Integer friendId) {
+        Optional<User> optionalFriend = userStorage.findAll().stream().filter(u -> u.getId() == friendId)
```

Comment:
 Можно упростить проверку с помощью `orElseThrow`
```suggestion
        User optionalFriend = userStorage.findAll().stream()
            .filter(u -> u.getId() == friendId)
            .findFirst()
            .orElseThrow(() -> new NonexistentException("user not existed"));
```